playbooks/robusta_playbooks/krr.py

```diff
@@ -379,31 +400,36 @@ def krr_scan(event: ExecutionBaseEvent, params: KRRParams):
         logs = logs[logs.find("{") :]
 
         krr_response = json.loads(logs)
-        end_time = datetime.now()
         krr_scan = KRRResponse(**krr_response)
-    except json.JSONDecodeError:
-        logging.exception("*KRR scan job failed. Expecting json result.*")
-        logging.error(f"Logs: {logs}")
-        return
-    except ValidationError:
-        logging.exception("*KRR scan job failed. Result format issue.*\n\n")
-        logging.error(f"Logs: {logs}")
-        return
+
     except Exception as e:
-        if str(e) == "Failed to reach wait condition":
+        if isinstance(e, json.JSONDecodeError):
+            logging.exception("*KRR scan job failed. Expecting json result.*")
+        elif isinstance(e, ValidationError):
+            logging.exception("*KRR scan job failed. Result format issue.*")
+        elif str(e) == "Failed to reach wait condition":
             logging.exception(f"*KRR scan job failed. The job wait condition timed out ({params.timeout}s)*")
         else:
             logging.exception(f"*KRR scan job unexpected error.*\n {e}")
+
         logging.error(f"Logs: {logs}")
+        update_state(ScanState.FAILED)
         return
+    else:
+        # TODO: Process the result, add some non-critical errors and warnings to the metadata
+        metadata["strategy"] = krr_scan.strategy.dict() if krr_scan.strategy else None
+        metadata["description"] = krr_scan.description
+        metadata["errors"] = krr_scan.errors
+        update_state(ScanState.SUCCESS)
```

Comment:
I agree, but I feel like it might be confusing to have state events only for pending and failed, but not for success.
Maybe if in the future someone will want to write handlers for a different sync - it will be confusing, that they do not receive success event

Reply:
I still think this is problematic, because it "breaks" the protocol
What I mean by that, is that until now, the meta table is updated only after all scan records were inserted
Now, there will be time, where the meta is marked as `success`, but not all (or none) results of the scan are in the DB

I think it's best to update it to success, only after all scan results were inserted (the ui doesn't have another way to know it's all there)